### DISCUSSION TITLE
Support for wide gamut contrast

### DISCUSSION THREAD

**Discussion by VojtechVidra** [Myndex/SAPC-APCA#98] [Ideas]
I'm looking for a way to check APCA contrast of wide gamut colors like LCH or XYZ. I see there is `displayP3toY` and `adobeRGBtoY` which are both wider gamut then sRGB but LCH is even wider so it would't be always 100% accurate.

I wasn't able to find what I'm looking for both in `apca-w3` and `colorparsley` packages. Do you have some suggestion on how to do this?

Thanks!

**Comment by Myndex** [comment]
Hi @VojtechVidra 

I'm sorry for the delay in reply, I must have missed seeing your message.

Thus far, we've only released input modules for real color spaces, that is, color spaces that relate to real, physical displays.

XYZ, CIELAB, CIELUV, LChab, LShuv, ACES, ProPhoto, and so forth are not "real" spaces, use imaginary primaries, and do not relate to actual, physical displays (ICC profiles not withstanding).

APCA is a simplified, perceptually uniform model for predicting contrast of high spatial frequency elements on self-illuminating displays (i.e. text on your screen). Because it is a perception model, the input needs to be either:
1. a transform from real (non-imaginary) color data, or
2. a transform from a perceptual space, i.e. a color appearance model.
    - APCA seems to work well with [HSLuv.org](https://HSLuv.org), which is a version of CIELUV
    - Some are having success integrating APCA into OKLab. See _Evil Martins_.
    - SACAM (Sjuv Accessible Color Appearance Model) contains APCA, also not yet released.

I hope this answers your question please feel free to follow up.
